CodeEdit/Features/CodeEditUI/Views/ToolbarBranchPicker.swift

```diff
@@ -71,8 +71,10 @@ struct ToolbarBranchPicker: View {
             isHovering = active
         }
         .onReceive(NotificationCenter.default.publisher(for: NSApplication.didBecomeActiveNotification)) { (_) in
-            Task {
-                await sourceControlManager?.refreshCurrentBranch()
+            if self.currentBranch != nil {
```

Comment:
I don't think so because `currentBranch` is cleared out when `sourceControlIsEnabled` is `false`. I could do the check anyways for clarity but that may be redundant.